api/test/unit/utils/test_watermark.py

```diff
@@ -14,55 +12,33 @@
 _MOCK_IMAGE_INFO = json.loads((_MOCK_IMAGE_PATH / "sample-image-info.json").read_text())
 
 
-@dataclass
-class RequestsFixture:
-    requests: list[Request]
-    response_factory: Callable[[Request], Response] = (  # noqa: E731
-        lambda x: RequestsFixture._default_response_factory(x)
+@pytest.fixture
+def mock_request():
+    pook.on()
+    mock = (
+        pook.get("http://example.com/")
+        .header("User-Agent", HEADERS["User-Agent"])
+        .reply(200)
+        .body(_MOCK_IMAGE_BYTES, binary=True)
+        .mock
     )
+    yield mock
+    pook.off()
```

Comment:
```suggestion
def mock_request():
    with pook.use():
        mock = (
            pook.get("http://example.com/")
            .header("User-Agent", HEADERS["User-Agent"])
            .reply(200)
            .body(_MOCK_IMAGE_BYTES, binary=True)
            .mock
        )
        yield mock
```

It's unlikely, but if creating the mock raised an exception then `pook.on` could be left on, never cleaned up.